Grasshopper_UI/Components/oM/CreateObject2.cs

```diff
@@ -0,0 +1,120 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2018, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Grasshopper.Kernel;
+using BH.UI.Grasshopper.Templates;
+using BH.UI.Templates;
+using BH.UI.Components;
+using Grasshopper.Kernel.Parameters;
+using Grasshopper.Kernel.Parameters.Hints;
+using BH.Engine.Grasshopper;
+
+namespace BH.UI.Grasshopper.Components
+{
+    public class CreateObject2Component : CallerComponent, IGH_VariableParameterComponent
+    {
+        /*******************************************/
+        /**** Properties                        ****/
+        /*******************************************/
+
+        public override Caller Caller { get; } = new CreateObject2Caller();
+
+
+        /*******************************************/
+        /**** Constructors                      ****/
+        /*******************************************/
+
+        public CreateObject2Component() : base()
+        {
+            this.Params.ParameterChanged += OnGrasshopperUpdates;
+        }
+
+
+        /*******************************************/
+        /**** Override Methods                  ****/
+        /*******************************************/
+
+        public override bool CanInsertParameter(GH_ParameterSide side, int index)
+        {
+            return side == GH_ParameterSide.Input;
```

Comment:
If you select a `Create` method from the menu, the component still leaves the possibility to remove and add parameters. If you remove one it crashes. Maybe we can prevent it from happening?
```suggestion
            return side == GH_ParameterSide.Input && !(Caller?.SelectedItem is MethodInfo);
```

Reply:
Thanks. Yes, that was on the todo list for today but good that you are picking this up.
Now I can just copy-paste 😄